src/web3/utils.ts

```diff
@@ -1,4 +1,35 @@
 import algosdk from "algosdk";
+import { TriggerBase, TriggerInit } from "grindery-nexus-common-utils";
+
+export type NewTransactionInput = { chain: string | string[]; from?: string; to?: string };
+export type NewEventInput = {
+  chain: string | string[];
+  contractAddress?: string;
+  eventDeclaration: string | string[];
+  parameterFilters: { [key: string]: unknown };
+};
+
+export type NewTransactionFlowInput = {
+  chain: string;
+  contract: string;
+  eventDeclaration: string;
+  from?: string;
+  to?: string;
+};
+
+export type TriggerBasePayload = Record<string, unknown>;
+export type TriggerBaseState = Record<string, unknown>;
```

Comment:
@SAPikachu I just removed the `TriggerBasePayload` and `TriggerBaseState` types whenever there is a definition of the `TriggerBase` class so that I now have `TriggerBase<T>`.

On the other hand, in my opinion, these types should not be removed when it comes to the constructor of the class, so as to have for example `TriggerInit<NewTransactionInput, TriggerBasePayload, TriggerBaseState>`

Reply:
> On the other hand, in my opinion, these types should not be removed when it comes to the constructor of the class, so as to have for example TriggerInit<NewTransactionInput, TriggerBasePayload, TriggerBaseState>
Any reason for this? I don't see any usefulness in them. Since different triggers send different payload and likely save different state, we can't use the same type for all of them anyways.